### PR TITLE
Add support for engine_exchangeCapabilities

### LINKED ISSUE #11921: Engine API - engine_exchangeCapabilities
# 💎 Issue

### Background

support sending engine api versions supported by Prysm in exchange receive engine APIs supported by the EL. This will make consensus and collaboration easier going forward to decide how to handle different versions prior to hard forks that require engine api changes.

https://github.com/ethereum/execution-apis/pull/364

### PR DESCRIPTION
**Pull request by saolyn**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

This PR adds [engine_exchangecapabilities](https://github.com/ethereum/execution-apis/blob/main/src/engine/common.md#engine_exchangecapabilities). Which is used to exchange a list of CL/EL supported engine methods.

**Which issues(s) does this PR fix?**

Fixes #11921

**Other notes for review**

### CODE REVIEW

File: beacon-chain/execution/engine_client.go [hunk 1]
```diff
@@ -30,6 +30,20 @@ import (
 	"go.opencensus.io/trace"
 )
 
+var (
```
Comment (rkapka) [inline beacon-chain/execution/engine_client.go:33]: I guess we should also include `ExchangeCapabilities` itself.

File: beacon-chain/execution/engine_client.go [hunk 2]
```diff
@@ -278,6 +294,35 @@ func (s *Service) ExchangeTransitionConfiguration(
...
+			if s1 == s2 {
+				supported = true
+				break
+			}
+		}
+		if !supported {
+			unsupported = append(unsupported, s1)
+		}
+	}
+	if len(unsupported) != 0 {
```
Comment (rkapka) [inline beacon-chain/execution/engine_client.go:320]: A unit test checking the log would be nice

File: beacon-chain/execution/engine_client.go [hunk 3]
```diff
@@ -278,6 +294,35 @@ func (s *Service) ExchangeTransitionConfiguration(
...
+		log.Warnf("Please update, detected the following unsupported engine methods: %s", unsupported)
```
Comment (rkapka) [inline beacon-chain/execution/engine_client.go:321]: `Please update` is too vague. Update what?

File: beacon-chain/execution/engine_client.go (hunk 1 above)
Comment (saolyn) [reply beacon-chain/execution/engine_client.go:33]: In the spec it says "The engine_exchangeCapabilities method MUST NOT be returned in the response list.". So I removed it from the list of our own supported ones as well. 
Would it be better to include it?

**Review by rkapka — APPROVED** [review]

**Pull request merged**